Compilation-Alpine-Linux.md

```diff
@@ -1,72 +1,69 @@
-
-The majority of these commands are copy and paste but some can be modified. For example, in the libtorrent section there is a choice between libtorrent `v1` or `v2`. Check the notes and commented commands for more information.
-
-Check out the [common information](https://github.com/qbittorrent/qBittorrent/wiki/Compilation-with-CMake-common-information) page to learn more about the available build configuration options for qBittorrent (for example, to compile qBittorrent without the GUI) and also CMake itself, if you're new to it.
+> [!IMPORTANT]
+> The majority of these commands are copy and paste but some can be modified. For example, in the libtorrent section there is a choice between libtorrent `v1` or `v2`. Check the notes and commented commands for more information.
 
 ## Build dependencies
 
 These are the build dependencies we need to install using `apk`
 
-```ash
-apk add autoconf automake build-base cmake curl git libtool linux-headers perl pkgconf python3 python3-dev re2c tar
+```bash
+apk add build-base cmake curl git linux-headers python3 re2c tar xz ninja-build ninja-is-really-ninja
 ```
 
 ## Application dependencies
 
 These are application dependencies we need to install using `apk`
 
-ℹ️ The `iconv` libraries needed for `libtorrent` v1.2 are built into `/usr/lib/libc.so` so we do not need to install `iconv` as a dependency.
-
-```ash
-apk add icu-dev libexecinfo-dev openssl-dev qt5-qtbase-dev qt5-qttools-dev zlib-dev qt5-qtsvg-dev
-```
-
-## Ninja build
-
-⚠️ There is no `apk` version for Alpine `3.12` or greater so we need to build it. It won't take long.
+> [!WARNING]
+> If you are using a desktop and want the GUI for qBittorrent, you will need to append the dependency `qt6-qtsvg-dev` to the command below
 
 ```bash
-git clone --shallow-submodules --recurse-submodules https://github.com/ninja-build/ninja.git ~/ninja && cd ~/ninja
-git checkout "$(git tag -l --sort=-v:refname "v*" | head -n 1)"
-cmake -Wno-dev -B build \
-	-D CMAKE_CXX_STANDARD=17 \
-	-D CMAKE_INSTALL_PREFIX="/usr/local"
-cmake --build build
-cmake --install build
+apk add zlib-dev icu-dev openssl-dev qt6-qtbase-dev qt6-qttools-dev
 ```
 
 ## Boost build files
 
-ℹ️  All we need to bootstrap boost is to download and extract the files. There is nothing to build at this step.
+> [!TIP]
+> Get the latest version info via
+>
+> - https://github.com/boostorg/boost/releases/latest
+> - `curl -sL https://api.github.com/repos/boostorg/boost/releases/latest | jq -r '.tag_name'`
 
 Download and extract the boost files.
 
+> [!NOTE]
+> All we need to bootstrap boost is to download and extract the files. There is nothing to build at this step.
+
 ```bash
-curl -sNLk https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.gz -o "$HOME/boost_1_76_0.tar.gz"
-tar xf "$HOME/boost_1_76_0.tar.gz" -C "$HOME"
+curl -sNLko- https://github.com/boostorg/boost/releases/latest/download/boost-1.86.0-b2-nodocs.tar.xz | tar xfJ -
```

Comment:
It would be best to keep these steps separated. So the user won't need to redownload when extraction goes wrong.